carma-messenger-core/j2735_convertor/src/control_message_convertor.cpp

```diff
@@ -246,7 +246,12 @@ void convert(const j2735_v2x_msgs::msg::TrafficControlSchedule& in_msg, carma_v2
   out_msg.end_exists = in_msg.end_exists;
   if(out_msg.end_exists)
   {
-    out_msg.end = rclcpp::Time(in_msg.end * units::SEC_PER_MIN, 0);
+    
+    // Obtained from carma_v2x_msgs/msg/TrafficControlSchedule.msg
+    double endtime_max =  153722867280912;
+    auto end_time = std::min(in_msg.end* units::SEC_PER_MIN, endtime_max);
```

Comment:
The use of 153722867280912 as the max here seems error prone.
According to the message spec, that is the maximum value of min in the epoch, but we are converting it to seconds which will be 153722867280912 * 60.
So it seems like 153722867280912 / 60 would exceed this value and be potentially incorrectly capped. Also this logic has no consideration for the actual cap of the relevant constructor.
A better approach would be to consider the type in question 
The rclcpp::Time(sec, nsec) constructor has a seconds field of type int32_t 
So we can use std::numeric_limits to find the true supported max value

```
#include <limits> // Add to top of file
std::numeric_limits<int32_t>::max();
```
Specifically
```
std::min(in_msg.end* units::SEC_PER_MIN, std::numeric_limits<int32_t>::max(););

Reply:
I believe the maximum value I used here is in seconds, but I modified according to your comment